### DISCUSSION TITLE
Paginating preloaded relationships

### DISCUSSION THREAD

**Discussion by arthur-clq** [adonisjs/core#3303] [Help] [answered]
Hi, is there a way to paginate relationship during preload?

Example: When a user enters his/her dashboard they see their account info and a list of their posts

```
const user = await User.query().where('id', id).preload('posts', (posts) => {
    posts.paginate(1, 6)
}).firstOrFail()
```

Currently this code gives me a "Cannot paginate relationship during preload" error and I came up with this way to get the results I want:

```
const user = await User.query().where('id', id).firstOrFail()
user.$extras.posts = await user.related('posts').query().paginate(1, 6)
```

and then I serialize `$extras`. I feel this is not an elegant solution.

Any help or discussion is welcomed. Thanks.

**Comment by thetutlage** [comment]
> Currently this code gives me a "Cannot paginate relationship during preload" error and I came up with this way to get the results I want:

It is true, coz I have never seen a UI where nested resources are paginated. You always paginate between the top level resources and hence paginated nested data is not supported

**Reply by tmd-billing** [reply]
Just Support Limit , but not support Limit .. 

for more reading https://docs.adonisjs.com/guides/models/relationships

**Comment by davidharting** [comment, marked as answer]
In your example in the original discussion, I would query for Posts directly, and filter by User ID in the where clause. 

That way, the thing you want to paginate on becomes the top level resource in the query.
